To investigate:

```
Traceback (most recent call last):
  File "./wpt", line 21, in <module>
    wpt.main()
  File "/Users/runner/work/1/s/tools/wpt/wpt.py", line 173, in main
    rv = script(*args, **kwargs)
  File "/Users/runner/work/1/s/tools/wpt/run.py", line 838, in run
    rv = run_single(venv, **wptrunner_kwargs) > 0
  File "/Users/runner/work/1/s/tools/wpt/run.py", line 845, in run_single
    return wptrunner.start(**kwargs)
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/wptrunner.py", line 402, in start
    rv = not run_tests(**kwargs) or logged_critical.has_log
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/wptrunner.py", line 190, in run_tests
    **kwargs)
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/wptrunner.py", line 92, in get_loader
    chunker_kwargs=chunker_kwargs)
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/testloader.py", line 247, in __init__
    self._load_tests()
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/testloader.py", line 310, in _load_tests
    for test_path, test_type, test in self.iter_tests():
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/testloader.py", line 303, in iter_tests
    yield test_path, test_type, self.get_test(manifest_file, test, inherit_metadata, test_metadata)
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/testloader.py", line 263, in get_test
    return wpttest.from_manifest(manifest_file, manifest_test, inherit_metadata, test_metadata)
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/wpttest.py", line 660, in from_manifest
    return test_cls.from_manifest(manifest_file, manifest_test, inherit_metadata, test_metadata)
  File "/Users/runner/work/1/s/tools/wptrunner/wptrunner/wpttest.py", line 502, in from_manifest
    **cls.cls_kwargs(manifest_test))
TypeError: __init__() got an unexpected keyword argument 'subdomain'
```

--- a/tools/manifest/item.py
+++ b/tools/manifest/item.py
@@ -147,6 +147,14 @@ class URLManifestItem(ManifestItem):
         flags = set(urlparse(self.url).path.rsplit("/", 1)[1].split(".")[1:-1])
         return "h2" in flags
 
+    @property
+    def subdomain(self):
+        # type: () -> bool
+        flags = set(urlparse(self.url).path.rsplit("/", 1)[1].split(".")[1:-1])
+        # Note: this is currently hard-coded to check for `www`, rather than
+        # all possible valid subdomains. It can be extended if needed.
+        return "www" in flags
+
     def to_json(self):
         # type: () -> Tuple[Optional[Text], Dict[Any, Any]]
         rel_url = None if self._url == self.path.replace(os.path.sep, u"/") else self._url

--- a/tools/manifest/tests/test_item.py
+++ b/tools/manifest/tests/test_item.py
@@ -47,6 +47,37 @@ def test_url_not_https(path):
     assert m.https is False
 
 
+@pytest.mark.parametrize("path", [
+    "a.www.c",
+    "a.b.www.c",
+    "a.www.b.c",
+    "a.b.www.c.d",
+    "a.https.www.c",
+    "a.b.https.www.c",
+    "a.https.www.b.c",
+    "a.b.https.www.c.d",
+])
+def test_url_subdomain(path):
+    m = HarnessTest("/foo", "bar/" + path, "/", "bar/" + path)
+
+    assert m.subdomain is True
+
+
+@pytest.mark.parametrize("path", [
+    "www",
+    "a.www",
+    "a.b.www",
+    "www.a",
+    "www.a.b",
+    "a.bwwww.c",
+    "a.wwwwb.c",
+])
+def test_url_not_subdomain(path):
+    m = HarnessTest("/foo", "bar/" + path, "/", "bar/" + path)
+
+    assert m.subdomain is False
+
+
 @pytest.mark.parametrize("fuzzy", [
     {('/foo/test.html', u'/foo/ref.html', '=='): [[1, 1], [200, 200]]},
     {('/foo/test.html', u'/foo/ref.html', '=='): [[0, 1], [100, 200]]},

--- a/tools/wptrunner/wptrunner/executors/base.py
+++ b/tools/wptrunner/wptrunner/executors/base.py
@@ -307,14 +307,18 @@ class TestExecutor(object):
 
         self.runner.send_message("test_ended", test, result)
 
-    def server_url(self, protocol):
+    def server_url(self, protocol, subdomain=False):
         scheme = "https" if protocol == "h2" else protocol
-        return "%s://%s:%s" % (scheme,
-                               self.server_config["browser_host"],
-                               self.server_config["ports"][protocol][0])
+        host = self.server_config["browser_host"]
+        if subdomain:
+            # The only supported subdomain filename flag is "www".
+            host = "{subdomain}.{host}".format(subdomain="www", host=host)
+        return "{scheme}://{host}:{port}".format(scheme=scheme, host=host,
+            port=self.server_config["ports"][protocol][0])
 
     def test_url(self, test):
-        return urljoin(self.server_url(test.environment["protocol"]), test.url)
+        return urljoin(self.server_url(test.environment["protocol"],
+                                       test.subdomain), test.url)
 
     @abstractmethod
     def do_test(self, test):

--- a/tools/wptrunner/wptrunner/wpttest.py
+++ b/tools/wptrunner/wptrunner/wpttest.py
@@ -158,13 +158,15 @@ class Test(object):
     long_timeout = 60  # seconds
 
     def __init__(self, tests_root, url, inherit_metadata, test_metadata,
-                 timeout=None, path=None, protocol="http", quic=False):
+                 timeout=None, path=None, protocol="http", subdomain=False,
+                 quic=False):
         self.tests_root = tests_root
         self.url = url
         self._inherit_metadata = inherit_metadata
         self._test_metadata = test_metadata
         self.timeout = timeout if timeout is not None else self.default_timeout
         self.path = path
+        self.subdomain = subdomain
         self.environment = {"protocol": protocol, "prefs": self.prefs, "quic": quic}
 
     def __eq__(self, other):
@@ -190,7 +192,8 @@ class Test(object):
                    test_metadata,
                    timeout=timeout,
                    path=os.path.join(manifest_file.tests_root, manifest_item.path),
-                   protocol=server_protocol(manifest_item))
+                   protocol=server_protocol(manifest_item),
+                   subdomain=manifest_item.subdomain)
 
     @property
     def id(self):
@@ -395,9 +398,9 @@ class TestharnessTest(Test):
 
     def __init__(self, tests_root, url, inherit_metadata, test_metadata,
                  timeout=None, path=None, protocol="http", testdriver=False,
-                 jsshell=False, scripts=None, quic=False):
+                 jsshell=False, scripts=None, subdomain=False, quic=False):
         Test.__init__(self, tests_root, url, inherit_metadata, test_metadata, timeout,
-                      path, protocol, quic)
+                      path, protocol, subdomain, quic)
 
         self.testdriver = testdriver
         self.jsshell = jsshell
@@ -422,6 +425,7 @@ class TestharnessTest(Test):
                    testdriver=testdriver,
                    jsshell=jsshell,
                    scripts=scripts,
+                   subdomain=manifest_item.subdomain,
                    quic=quic)
 
     @property
@@ -453,8 +457,8 @@ class ReftestTest(Test):
     test_type = "reftest"
 
     def __init__(self, tests_root, url, inherit_metadata, test_metadata, references,
-                 timeout=None, path=None, viewport_size=None, dpi=None, fuzzy=None, protocol="http",
-                 quic=False):
+                 timeout=None, path=None, viewport_size=None, dpi=None, fuzzy=None,
+                 protocol="http", quic=False):
         Test.__init__(self, tests_root, url, inherit_metadata, test_metadata, timeout,
                       path, protocol, quic)
 

